src/pyitemmodel_proxy.cpp

```diff
@@ -0,0 +1,232 @@
+    if (!proxy->m_qpythonitemmodel_ref) {
+        qFatal("proxy no longer exists");  // XXX handle with exception
+        return NULL;  // never gets here
+    }
+    QPythonItemModel *model = (QPythonItemModel *)proxy->m_qpythonitemmodel_ref->value();
+    model->sendDataChanged(args);
+    Py_RETURN_NONE;
+}
+
+PyObject *
+beginInsertRows(PyObject *self, PyObject *args)
+{
+    pyotherside_QPythonItemModelProxy *proxy = (pyotherside_QPythonItemModelProxy *)self;
+    if (!proxy->m_qpythonitemmodel_ref) {
+        qFatal("proxy no longer exists");  // XXX handle with exception
+        return NULL;  // never gets here
+    }
+    QPythonItemModel *model = (QPythonItemModel *)proxy->m_qpythonitemmodel_ref->value();
+    PyObject *parent_path;
+    int first, last;
+    if (!PyArg_ParseTuple(args, "Oii", &parent_path, &first, &last))
+	return NULL;  // XXX failure here doesn't lead to traceback for some reason
+    QModelIndex parent(model->pysequence_to_qmodelindex(parent_path));
+    model->do_beginInsertRows(parent, first, last);
+    Py_RETURN_NONE;
+}
+
+PyObject *
+beginRemoveRows(PyObject *self, PyObject *args)
+{
+    pyotherside_QPythonItemModelProxy *proxy = (pyotherside_QPythonItemModelProxy *)self;
+    if (!proxy->m_qpythonitemmodel_ref) {
+        qFatal("proxy no longer exists");  // XXX handle with exception
+        return NULL;  // never gets here
+    }
+    QPythonItemModel *model = (QPythonItemModel *)proxy->m_qpythonitemmodel_ref->value();
+    PyObject *parent_path;
+    int first, last;
+    if (!PyArg_ParseTuple(args, "Oii", &parent_path, &first, &last))
+	return NULL;  // XXX failure here doesn't lead to traceback for some reason
+    QModelIndex parent(model->pysequence_to_qmodelindex(parent_path));
+    model->do_beginRemoveRows(parent, first, last);
+    Py_RETURN_NONE;
+}
+
+PyObject *
+endInsertRows(PyObject *self, PyObject *null)
+{
+    pyotherside_QPythonItemModelProxy *proxy = (pyotherside_QPythonItemModelProxy *)self;
+    if (!proxy->m_qpythonitemmodel_ref) {
+        qFatal("proxy no longer exists");  // XXX handle with exception
+        return NULL;  // never gets here
+    }
+    QPythonItemModel *model = (QPythonItemModel *)proxy->m_qpythonitemmodel_ref->value();
+    model->do_endInsertRows();
+    Py_RETURN_NONE;
+}
+
+PyObject *
+endRemoveRows(PyObject *self, PyObject *null)
+{
+    pyotherside_QPythonItemModelProxy *proxy = (pyotherside_QPythonItemModelProxy *)self;
+    if (!proxy->m_qpythonitemmodel_ref) {
+        qFatal("proxy no longer exists");  // XXX handle with exception
+        return NULL;  // never gets here
+    }
+    QPythonItemModel *model = (QPythonItemModel *)proxy->m_qpythonitemmodel_ref->value();
+    model->do_endRemoveRows();
+    Py_RETURN_NONE;
+}
+
+PyMethodDef pyotherside_QPythonItemModelProxyType_methods[] = {
+    {"call_back_for_modification", (PyCFunction)call_back_for_modification, METH_O,
```

Comment:
I've implemented this inside PythonItemModelProxy for now. But perhaps it's a more generic thing for a Python thread to want to do something in the UI thread, and thus it should be a generic method available through the `pyotherside` module?